lib/StripeResource.js

```diff
@@ -394,15 +413,27 @@ StripeResource.prototype = {
 
       req.once('response', (res) => {
         if (this._shouldRetry(res, requestRetries)) {
-          return retryRequest(makeRequest, apiVersion, headers, requestRetries);
+          return retryRequest(
+            makeRequest,
+            apiVersion,
+            headers,
+            requestRetries,
+            ((res || {}).headers || {})['retry-after']
```

Comment:
This makes my head hurt a bit. Also won't it throw an error if  `headers` doesn't exist? Maybe something like:

`const retryAfter = res.hasOwnProperty('headers') ? res.headers['retry-after'] || null : null;`

Wish optional chaining was already in JS 😭

Reply:
That's funny.  I love this and it validates so many of my life decisions. 😂 

It won't throw because it takes either res or an empty object, and then either that thing or an empty object, and so eventually gets `retry-after` from either `res['headers']` or an empty object.

It's Javascript at is finest. 🥂

Reply:
heh, yeah, this is basically how I do optional chaining until it's a Real Feature.